fineract-provider/src/main/java/org/apache/fineract/portfolio/delinquency/service/DelinquencyReadPlatformServiceImpl.java

```diff
@@ -126,4 +134,20 @@ public CollectionData calculateLoanCollectionData(final Long loanId) {
         return collectionData;
     }
 
+    @Override
+    public Collection<LoanInstallmentDelinquencyTagData> retrieveLoanInstallmentsCurrentDelinquencyTag(Long loanId) {
```

Comment:
Please rewrite this to not load the loan (even if you fetch only a reference, then moment you are fetching one of its field, its fetches the full loan) and fetch one by one the installments and its delinquency tags, please do it in one shot:
- Write a query that fetches them in the delinquency repository

Reply:
done. Retrieving only DelinquencyRange and OutstandingAmount for installments as required by Api.